python/foglamp/services/core/server.py

```diff
@@ -340,15 +391,23 @@ def _start_core(cls, loop=None):
                 _logger.info('Loading certificates %s and key %s', cert, key)
                 ssl_ctx.load_cert_chain(cert, key)
 
+            # Get the service data and advertise the management port of the core to allow other microservices to find FogLAMP
+            loop.run_until_complete(cls.service_config())
+            _logger.info('Announce management API service')
+            cls.management_announcer = ServiceAnnouncer('core.{}'.format(cls._service_name), cls._MANAGEMENT_SERVICE, cls.core_management_port,
```

Comment:
Major concern: so by default if we deployed 100 (unattended) devices, all will have same name and type; unless configuration changed and REBOOTED!

I would prefer to use MAC address Or a uuid (saving it as config item will keep it same on reboot too) | and admin can always change it as needed to see something per their choice.

Reply:
I honestly think the use of UUID would be very confusing. 
The choice of the MAC address is better, but then we should use the IP address, which makes the whole concept of auto-discoverability pretty pointless.

Reply:
is mac address dynamic? the way IP is on DHCP network?

Reply:
Let's make it clear: there is no perfect solution here.
1. UUID: it changes every time you reboot FogLAMP: you should change name accordingly
2. MAC address: there are multiple mac addresses (a Raspberry PI already has 2!), you must arbitrary select one of them
3. IP address: slightly better than UUID, but also bad for the multiple interfaces.
4. Default name: you may have machines advertising the same name.

In all this, I think the easiest and simplest way of handling this for a sysadmin is #4. Scripts can be easily set up to handle it, and so we may do with GUIs. 

From a product perspective, let's close this PR with #4 please.

Reply:
Okay! I believe we could have picked any of #2 (to advertise better) but as decided let's go ahead as decided.

Reply:
For 4) we may, but that will not give expected results 
Please try `avahi-browse -d local _foglamp-manage._tcp --resolve -t`

Also on win/mac, we may want to use bonjour. hence proper txtrecord is good to have.

Reply:
I was thinking at a more simple solution, i.e. scripting for foglamp and the suffix added by avahi automatically. I honestly think we are good with this.